Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/presenter/ImageViewerPresenter.java

```diff
@@ -222,16 +244,21 @@ private void onPageSwipedTo(int position) {
             callback.setImageMode(other, LOWRES, false);
         }
 
-        cancelPreviousImageDownload(position);
+        synchronized (nonCancelableImages) {
+            nonCancelableImages.clear();
+            nonCancelableImages.addAll(getNonCancelableImages(position));
+        }
+
+        if (swipingForward) {
+            cancelPreviousFromStartImageDownload(position);
+        } else {
+            cancelPreviousFromEndImageDownload(position);
+        }
 
         // Already in LOWRES mode
         if (callback.getImageMode(postImage) == LOWRES) {
             onLowResInCenter();
         }
-        // Else let onModeLoaded handle it
-
-        callback.showProgress(progress.get(selectedPosition) >= 0f);
-        callback.onLoadProgress(progress.get(selectedPosition));
```

Comment:
I should actually probably use these two methods here but with default values so that we should progress bar immediatelly without having to wait the response from the server for a HEAD request.